services/ux-backend/handlers/onboarding/peertokens/handler.go

```diff
@@ -1,30 +1,73 @@
 package peertokens
 
 import (
+	"encoding/json"
 	"fmt"
 	"net/http"
+	"strings"
 
+	ocsv1 "github.com/red-hat-storage/ocs-operator/api/v4/v1"
 	"github.com/red-hat-storage/ocs-operator/v4/controllers/util"
 	"github.com/red-hat-storage/ocs-operator/v4/services/ux-backend/handlers"
 
 	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
 const (
 	onboardingPrivateKeyFilePath = "/etc/private-key/key"
 )
 
-func HandleMessage(w http.ResponseWriter, r *http.Request, tokenLifetimeInHours int) {
+func HandleMessage(w http.ResponseWriter, r *http.Request, tokenLifetimeInHours int, cl client.Client) {
 	switch r.Method {
 	case "POST":
-		handlePost(w, r, tokenLifetimeInHours)
+		handlePost(w, r, tokenLifetimeInHours, cl)
 	default:
 		handleUnsupportedMethod(w, r)
 	}
 }
 
-func handlePost(w http.ResponseWriter, _ *http.Request, tokenLifetimeInHours int) {
-	if onboardingToken, err := util.GeneratePeerOnboardingToken(tokenLifetimeInHours, onboardingPrivateKeyFilePath); err != nil {
+func handlePost(w http.ResponseWriter, r *http.Request, tokenLifetimeInHours int, cl client.Client) {
+	if r.ContentLength == 0 {
+		http.Error(w, "Request body is empty, expected StorageCluster NamespacedName", http.StatusBadRequest)
+		return
+	}
+
+	var body = struct {
+		StorageClusterNamespacedName string `json:"storageClusterNamespacedName"`
+	}{}
+	if err := json.NewDecoder(r.Body).Decode(&body); err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	parts := strings.Split(body.StorageClusterNamespacedName, "/")
+	if len(parts) != 2 || parts[0] == "" || parts[1] == "" {
+		http.Error(w,
+			fmt.Sprintf(
+				"storageCluster NamespacedName is not in expected format, expected: Namespace/Name. got %v",
+				body.StorageClusterNamespacedName,
+			),
+			http.StatusBadRequest,
+		)
+		return
+	}
+
+	storageCluster := &ocsv1.StorageCluster{}
+	storageCluster.Name = parts[1]
+	storageCluster.Namespace = parts[0]
+
+	err := cl.Get(r.Context(), client.ObjectKeyFromObject(storageCluster), storageCluster)
+	if err != nil {
+		http.Error(
+			w,
+			fmt.Sprintf("failed to find StorageCluster with NamespacedName %s: %v", body.StorageClusterNamespacedName, err),
+			http.StatusBadRequest,
+		)
+		return
+	}
```

Comment:
This entire logic here needs to be duplicated to the client onboarding token handler. I would suggest to have this as a util of sorts to be invoked from both places